azure/test/client.test.js

```diff
@@ -292,6 +292,20 @@ describe('Azure Log Monitoring', function() {
                 this.forwarder.extractMetadataFromResource(record)
             );
         });
+        it('should correctly parse provider-only resource ids', function() {
+            record = {
+                resourceId:
+                    '/SUBSCRIPTIONS/12345678-1234-ABCD-1234-1234567890AB/PROVIDERS/MICROSOFT.RECOVERYSERVICES/SOMETHING/SOMETHINGELSE'
```

Comment:
Another useful test case would be a resource id with `microsoft.` in the actual resource name, to ensure we don't change customer-made names.

Reply:
We don't actually change the resource id, we just use the part of the resource ID with the `microsoft.` in it to determine the source, and add that to the metadata replacing `microsoft.` with `azure.`

Reply:
Or do you mean when parsing out the source that we don't grab an incorrect field?

Reply:
Yes mostly just parsing out the correct field. It's just a test case I like to include when I write code with a .replace in it.